spec/condorcet_poll_spec.rb

```diff
@@ -0,0 +1,110 @@
+      end
+
+      it 'raises InvalidCandidateError' do
+        poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob'], DateTime.now + 10)
+        vote = RankedVote.new('Miyoshi', ['Bob', 'Bob'])
+
+        expect { poll.add_vote(vote) }.to raise_error CondorcetPoll::InvalidCandidateError
+      end
+    end
+
+    context 'with a vote that has an invalid voter' do
+      it 'raises MultipleVoteError' do
+        poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob'], DateTime.now + 10)
+        vote = RankedVote.new('Miyoshi', ['Alice', 'Bob'])
+
+        poll.add_vote(vote)
+        expect { poll.add_vote(vote) }.to raise_error CondorcetPoll::MultipleVoteError
+      end
+    end
+
+    context 'with a vote that is overdue' do
+      it 'raises OverdueVoteError' do
+        poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob'], DateTime.now - 10)
+        vote = RankedVote.new('Miyoshi', ['Alice', 'Bob'])
+
+        expect { poll.add_vote(vote) }.to raise_error CondorcetPoll::OverdueVoteError
+      end
+    end
+  end
+
+  describe '#count_votes' do
+    it 'count the votes and returns the result as a winner array' do
+      poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob'], DateTime.now + 10)
+      poll.add_vote(RankedVote.new('Carol', ['Alice', 'Bob']))
+      poll.add_vote(RankedVote.new('Dave', ['Alice', 'Bob']))
+      poll.add_vote(RankedVote.new('Ellen', ['Bob', 'Alice']))
+
+      expect(poll.count_votes).to eq ['Alice']
+    end
+
+    it 'count the votes and returns the result as a winner array' do
+      poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob'], DateTime.now + 10)
+      poll.add_vote(RankedVote.new('Carol', ['Bob', 'Alice']))
+      poll.add_vote(RankedVote.new('Dave', ['Bob', 'Alice']))
+      poll.add_vote(RankedVote.new('Ellen', ['Bob', 'Alice']))
+
+      expect(poll.count_votes).to eq ['Bob']
+    end
+
+    it 'count the votes and returns the result as a winner array' do
+      poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob', 'Carol'], DateTime.now + 10)
+      poll.add_vote(RankedVote.new('Dave', ['Alice', 'Bob', 'Carol']))
+      poll.add_vote(RankedVote.new('Ellen', ['Carol', 'Alice', 'Bob']))
+      poll.add_vote(RankedVote.new('Frank', ['Carol', 'Alice', 'Bob']))
+
+      expect(poll.count_votes).to eq ['Carol']
+    end
+
+    it 'count the votes and returns the result as a winner array' do
+      poll = CondorcetPoll.new('Awesome Poll', ['Alice', 'Bob', 'Carol'], DateTime.now + 10)
+      poll.add_vote(RankedVote.new('Dave', ['Alice', 'Bob', 'Carol']))
+      poll.add_vote(RankedVote.new('Ellen', ['Carol', 'Alice', 'Bob']))
+      poll.add_vote(RankedVote.new('Frank', ['Bob', 'Carol', 'Alice']))
+
+      expect(poll.count_votes).to include 'Alice'
+      expect(poll.count_votes).to include 'Bob'
+      expect(poll.count_votes).to include 'Carol'
+      expect(poll.count_votes.length).to eq 3
```

Comment:
`#count_votes` の結果は同じなのに呼ぶたびに勝者計算が走って無駄なので、結果を変数に入れるとよいです。
また、要素の順番を問わずに配列の要素を検証したい場合は [`contain_exactly`](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher) という matcher があります。

```suggestion
      winners = poll.count_votes

      expect(winners).to contain_exactly('Alice', 'Bob', 'Carel')
```

Reply:
あ～～これも呼び出し演算子ぽいものがないので関数呼出を意識できていないパターンです。ありがとうございます。